util/mkconf.c

```diff
@@ -0,0 +1,142 @@
+  { "bsd-fatal",        "BSD_FATAL",        "Exit application on BSD-socket fatal errors" },
+  { "bootp",            "BOOTP",            "Include BOOTP client code" },
+  { "dhcp",             "DHCP",             "Include DHCP boot client code" },
+  { "rarp",             "RARP",             "Include RARP boot client code.  Contributed by Dan Kegel." },
+  { "geoip",            "GEOIP",            "Include GeoIP support. From Tor's geoip.c" },
+  { "ipv6",             "IPV6",             "Include IPv6 dual-stack support" },
+  { "language",         "LANGUAGE",         "Include Language translation code" },
+  { "fragments",        "FRAGMENTS",        "Include IP-fragment handling" },
+  { "statistics",       "STATISTICS",       "Include protocol statistics count" },
+  { "stackwalker",      "STACKWALKER",      "Use StackWalker for allocations and crash-dumps." },
+  { "fsext",            "FSEXT",            "Use djgpp's File Extensions for file I/O functions" },
+  { "loopback",         "LOOPBACK",         "Use the simple loopback device" },
+  { "embedded",         "EMBEDDED",         "Make an embeddable (ROM-able) target.  See note in config.h" },
+  { "buffered-io",      "BUFFERED_IO",      "Use buffered file I/O in pcconfig.c.  Otherwise use a simple read-ahead cache" },
+  { "tftp",             "TFTP",             "Include TFTP protocol for simple file retrival" },
+  { "udp-only",         "UDP_ONLY",         "Exclude all stuff related to the TCP protocol" },
+  { "tcp-sack",         "TCP_SACK",         "Include TCP Selective ACK feature (not yet)" },
+  { "echo-disc",        "ECHO_DISC",        "Include echo/discard servers" },
+  { "pppoe",            "PPPOE",            "PPP-over-Ethernet encapsulation" },
+  { "fast-pkt",         "FAST_PKT",         "Use faster all real-mode PKTDRVR receiver (DOSX only)" },
+  { "dead-gwd",         "DEAD_GWD",         "Dead Gateway detection in ARP module" },
+  { "gzip",             "GZIP",             "Use gzip compression for PPP and/or pcap debug file.  Also for SCTP." },
+  { "profiler",         "PROFILER",         "Include simple execution profiler (only DOSX with 64-bit types)" },
+  { "tcp-md5",          "TCP_MD5",          "TCP MD5 signature option" },
+  { "idna",             "IDNA",             "Support national characters in DNS name lookups." },
+  { "dynip-cli",        "DYNIP_CLI",        "Support DynDNS.org name/IP client" },
+  { "sctp",             "SCTP",             "Stream Control Transfer Protocol; experimental" }
+};
+
+#define NUM_OPTIONS (sizeof (options) / sizeof (struct cfgopt))
+
+static enum optval option_values[NUM_OPTIONS] = { };
+
+static int skip (const char* prefix, const char** arg)
+{
+  const size_t len = strlen (prefix);
+  if (strncmp (prefix, *arg, len) == 0)
+  {
+    *arg += len;
+    return (1);
+  }
+  return (0);
+}
+
+static void usage ()
+{
+  int i;
+
+  fprintf(stderr, "Available options:\n");
```

Comment:
Help-text like this should IMHO go to `stdout`.
Besides what does it look like now?

Reply:
OK, will change that.

The help text looks as follows:
```
> configur.bat --help
Configuring Watt-32 tcp/ip targets.
Usage: configur.bat [options...] targets...
Available targets:
  borland, clang, cygwin, djgpp, highc, mingw32, mingw64, orangec,
  pellesc, visualc, watcom, all, clean
Available options:
  --[en|dis]able-debug        Include debug code
  --[en|dis]able-multicast    Include IP multicast code by Jim Martin
  --[en|dis]able-bind         Include Bind resolver code
  --[en|dis]able-bsd-api      Include BSD-sockets
  --[en|dis]able-bsd-fatal    Exit application on BSD-socket fatal errors
  --[en|dis]able-bootp        Include BOOTP client code
  --[en|dis]able-dhcp         Include DHCP boot client code
  --[en|dis]able-rarp         Include RARP boot client code.  Contributed by Dan Kegel.
  --[en|dis]able-geoip        Include GeoIP support. From Tor's geoip.c
  --[en|dis]able-ipv6         Include IPv6 dual-stack support
  --[en|dis]able-language     Include Language translation code
  --[en|dis]able-fragments    Include IP-fragment handling
  --[en|dis]able-statistics   Include protocol statistics count
  --[en|dis]able-stackwalker  Use StackWalker for allocations and crash-dumps.
  --[en|dis]able-fsext        Use djgpp's File Extensions for file I/O functions
  --[en|dis]able-loopback     Use the simple loopback device
  --[en|dis]able-embedded     Make an embeddable (ROM-able) target.  See note in config.h
  --[en|dis]able-buffered-io  Use buffered file I/O in pcconfig.c.  Otherwise use a simple read-ahead cache
  --[en|dis]able-tftp         Include TFTP protocol for simple file retrival
  --[en|dis]able-udp-only     Exclude all stuff related to the TCP protocol
  --[en|dis]able-tcp-sack     Include TCP Selective ACK feature (not yet)
  --[en|dis]able-echo-disc    Include echo/discard servers
  --[en|dis]able-pppoe        PPP-over-Ethernet encapsulation
  --[en|dis]able-fast-pkt     Use faster all real-mode PKTDRVR receiver (DOSX only)
  --[en|dis]able-dead-gwd     Dead Gateway detection in ARP module
  --[en|dis]able-gzip         Use gzip compression for PPP and/or pcap debug file.  Also for SCTP.
  --[en|dis]able-profiler     Include simple execution profiler (only DOSX with 64-bit types)
  --[en|dis]able-tcp-md5      TCP MD5 signature option
  --[en|dis]able-idna         Support national characters in DNS name lookups.
  --[en|dis]able-dynip-cli    Support DynDNS.org name/IP client
  --[en|dis]able-sctp         Stream Control Transfer Protocol; experimental
```

Reply:
> The help text looks as follows:

Nice!

Reply:
Even nicer now, I made the text wrap around:
```
> configur.bat --help
Configuring Watt-32 tcp/ip targets.
Usage: configur.bat [options...] targets...
Available targets:
  borland, clang, cygwin, djgpp, highc, mingw32, mingw64, orangec,
  pellesc, visualc, watcom, all, clean
Available options:
  --[en|dis]able-debug        Include debug code.
  --[en|dis]able-multicast    Include IP multicast code by Jim Martin.
  --[en|dis]able-bind         Include Bind resolver code.
  --[en|dis]able-bsd-api      Include BSD-sockets.
  --[en|dis]able-bsd-fatal    Exit application on BSD-socket fatal errors.
  --[en|dis]able-bootp        Include BOOTP client code.
  --[en|dis]able-dhcp         Include DHCP boot client code.
  --[en|dis]able-rarp         Include RARP boot client code.  Contributed by
                              Dan Kegel.
  --[en|dis]able-geoip        Include GeoIP support.  From Tor's geoip.c.
  --[en|dis]able-ipv6         Include IPv6 dual-stack support.
  --[en|dis]able-language     Include Language translation code.
  --[en|dis]able-fragments    Include IP-fragment handling.
  --[en|dis]able-statistics   Include protocol statistics count.
  --[en|dis]able-stackwalker  Use StackWalker for allocations and crash-dumps.
  --[en|dis]able-fsext        Use djgpp's File Extensions for file I/O
                              functions.
  --[en|dis]able-loopback     Use the simple loopback device.
  --[en|dis]able-embedded     Make an embeddable (ROM-able) target.  See note
                              in config.h.
  --[en|dis]able-buffered-io  Use buffered file I/O in pcconfig.c.  Otherwise
                              use a simple read-ahead cache.
  --[en|dis]able-tftp         Include TFTP protocol for simple file retrival.
  --[en|dis]able-udp-only     Exclude all stuff related to the TCP protocol.
  --[en|dis]able-tcp-sack     Include TCP Selective ACK feature (not yet).
  --[en|dis]able-echo-disc    Include echo/discard servers.
  --[en|dis]able-pppoe        PPP-over-Ethernet encapsulation.
  --[en|dis]able-fast-pkt     Use faster all real-mode PKTDRVR receiver (DOSX
                              only).
  --[en|dis]able-dead-gwd     Dead Gateway detection in ARP module.
  --[en|dis]able-gzip         Use gzip compression for PPP and/or pcap debug
                              file.  Also for SCTP.
  --[en|dis]able-profiler     Include simple execution profiler (only DOSX with
                              64-bit types).
  --[en|dis]able-tcp-md5      TCP MD5 signature option.
  --[en|dis]able-idna         Support national characters in DNS name lookups.
  --[en|dis]able-dynip-cli    Support DynDNS.org name/IP client.
  --[en|dis]able-sctp         Stream Control Transfer Protocol; experimental.
```